src/sidebar/services/api.js

```diff
@@ -93,17 +90,16 @@ function get(object, path) {
  * @param {Promise<RouteMap>} links - API route data from API index endpoint (`/api/`)
  * @param {string} route - The dotted path of the named API route (eg. `annotation.create`)
  * @param {APIMethodCallbacks} callbacks
- * @return {APICallFunction}
+ * @return {APICall<Record<string, any>, object, object>}
```

Comment:
Lower in this module, the `APICall` usage is explained, but here it requires some back-and-forth with the type definition to understand what's going on. It would be nice if there could be a clear connection back to the types. e.g. `Record<string, any>` is `{Parems} params`, the first `object` is `{Body} data`, the second is `Promise<Result>`. It's mildly counterintuitive that the third...for lack of better terminology...item here defines a return type. I almost wonder if it might be worth using `@template` to define these params and return type, just for the extra clues it leaves. Then again, this is already fairly verbose. I mean, I get it. It just took an extra bit of thinking.

Reply:
I've added some notes here about the fact that the `APICall` here uses quite generic types and the caller is expected to cast to something more specific.